lib/dynamoid/persistence.rb

```diff
@@ -251,13 +251,33 @@ def undump_hash_value(val)
       end
 
       def format_datetime(value, options)
-        options[:store_as_native_string] ? value.iso8601 : value.to_time.to_f
+        use_string_format = options[:store_as_string].nil? \
+          ? Dynamoid.config.store_datetime_as_string \
+          : options[:store_as_string]
+
+        use_string_format ? value.to_time.iso8601 : value.to_time.to_f
+      end
+
+      def format_date(value, options)
+        use_string_format = options[:store_as_string].nil? \
+          ? Dynamoid.config.store_date_as_string \
+          : options[:store_as_string]
+
+        unless use_string_format
```

Comment:
Did you write it this way to optimize for fact that it'll likely be false?

Reply:
@richardhsu Yep. I consider numeric format as a common case and string one as an exception.